### ISSUE TITLE
Key generation field

### ISSUE THREAD

**Issue by Irismoon** [retorquere/zotero-better-bibtex#1306]
**Report ID:**

**Expected behavior:**

**Actual behavior:**

Hey, thanks for your awesome work. The tool is really handy and elegant. I just ask if it's possible to include a function: now the citekey use fixed number of fields. But if there is duplicate citekey, it 'll add a postfix like a,b,..., if it's possible to add additional optional fields like its title in addition to typical fields. It'll be easier to find the paper I want when I cite it. 
I check the wiki website. And find there is no similar functions in the citation key description.

Best,
Mango

**Comment by Irismoon** [comment]
It seems a suggestion rather than a bug. Apologize if wrong place.

**Comment by retorquere** [comment]
You can add the title or pretty much anything else you want to the citekey: https://retorque.re/zotero-better-bibtex/citing/ . What you're seeing now is just the default pattern but it's very configurable.

**Comment by Irismoon** [comment]
Hey thanks for reply. To be clear, what I mean is a special config when conflict occurs. For example, my config is [author][journal][year]. Usually the citekey is as above. But When there are two items have the same Citekey, is it possible to add an additional field to discriminate them more clearly? Like only under conflict case, the citekey could be  [author][journal][year][title].

Best,
Mango

**Comment by retorquere** [comment]
So you mean that if a conflict occurs, both would change?

**Comment by Irismoon** [comment]
Sorry for the delay due to time difference.
Sort of. Exactly, I mean if a conflict occurs, more fields are allowed to be added rather than a postfix like a,b,c,... Let's say now, I already have an item citekey is mattNature2019([author][journal][year]). When I add a new item whose citekey is automatically assigned as mattNature2019a as default, I hope both of them could change like one is mattNature2019July, mattNature2019Jan. The benefit is that when I cite them, I could discriminate them. I don't need to check mattNature2019 and mattNature2019a refers to which paper respectively. I of course know mattNature2019July and mattNature2019Jan refers to which paper because I read them and know their publication month.
To sum up, adding a postfix like a,b,... is an unknown information for me but the publication month is known information for me. I feel like it would help when citing.
But if there is no conflict at all, the citekey is still mattNature2019. 
Glad to hear how you think.

Best,
Mango

**Comment by retorquere** [comment]
That could trigger a cascade of updates. I'd rather not add that. 

I could consider a feature where you could:

1. Choose to not have keys postfixed
2. More easily find duplicates
3. A way to apply a custom pattern by right-clicking

Which should get your roughly the desired behavior, if not fully automatic.

**Comment by retorquere** [comment]
closing for inactivity
